src/test/java/org/jabref/logic/bibtex/comparator/FieldComparatorTest.java

```diff
@@ -173,4 +173,97 @@ public void compareStringFieldsBiggerDescending() throws Exception {
 
         assertEquals(1, comparator.compare(bigger, smaller));
     }
+
+    @Test
+    public void compareNumericFieldsIdentity() throws Exception {
+        FieldComparator comparator = new FieldComparator(StandardField.PMID);
+        BibEntry equal = new BibEntry();
+        equal.setField(StandardField.PMID, "123456");
```

Comment:
BibEntry equal = new BibEntry().withField(...)
```suggestion
        BibEntry equal = new BibEntry()
        	.withField(StandardField.PMID, "123456");
```

Reply:
Hi @calixtus 
Thank you for reviewing and for this suggestion!

I've applied it to the rest of the code (commit 25c47df).